### PR TITLE
Sub prometheus metrics differentiate between user and prow errors

### PR DESCRIPTION
**Pull request by chaodaiG**
/cc @cjwagner @listx

### CODE REVIEW

File: prow/pubsub/subscriber/metrics.go [hunk 1]
```diff
@@ -36,7 +39,7 @@ var (
 	errorCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
 		Name: "prow_pubsub_error_counter",
 		Help: "A counter of the webhooks made to prow.",
-	}, []string{subscriptionLabel})
+	}, []string{subscriptionLabel, errorTypeLabel})
```
Comment (cjwagner) [inline prow/pubsub/subscriber/metrics.go:42]: Do you know what will happen to the existing metric data that doesn't have this label?

File: prow/pubsub/subscriber/subscriber.go [hunk 2]
```diff
@@ -327,7 +331,10 @@ func (s *Subscriber) handleMessage(msg messageInterface, subscription string, al
 	eType, err := extractFromAttribute(msg.getAttributes(), prowEventType)
 	if err != nil {
 		l.WithError(err).Error("failed to read message")
-		s.Metrics.ErrorCounter.With(prometheus.Labels{subscriptionLabel: subscription})
+		s.Metrics.ErrorCounter.With(prometheus.Labels{
```
Comment (cjwagner) [inline prow/pubsub/subscriber/subscriber.go:334]: I don't think this actually increments the counter (and didn't previously either). We need to append a `.Inc()` call to actually use the counter that we've just indexed out of the vector. This applies to the other places as well.

File: prow/pubsub/subscriber/metrics.go (hunk 1 above)
Comment (chaodaiG) [reply prow/pubsub/subscriber/metrics.go:42]: I think I've seen it before, it's default to empty string when this field doesn't exist

File: prow/pubsub/subscriber/subscriber.go (hunk 2 above)
Comment (chaodaiG) [reply prow/pubsub/subscriber/subscriber.go:334]: good catch, fixed

**Review by cjwagner — APPROVED** [review]

**Pull request merged**